Examples/ScaleMicrogrid/ScaleMicrogrid.cpp

```diff
@@ -24,14 +24,19 @@ using real_type = double;
 #include "SolutionKeys.hpp"
 
 
-static int test(index_type Nsize, bool debug_output = false);
+static int test(index_type Nsize, real_type test_tolerance, bool error_tol = false, bool use_DAE_keys = false);
 
 int main(int argc, char const *argv[])
 {
     int retval = 0;
-    retval += test(2);
-    retval += test(4);
-    retval += test(8);
+    bool debug_out = false;
+    bool DAE_keys = false;
+
+    real_type error_tol = 8e-5;
+
```

Comment:
By the way, all tolerances should be defined in `SolutionKeys.hpp` file because they depend on how these values in the solution key are generated. I would implement it like:
 ```suggestion
    real_type error_tol = MICROGRID_TOL;
```
and define `MICROGRID_TOL` in the `SolutionKeys.hpp` file. There should be an explanation why you picked `8e-5` and not `7e-5`, for example. The explanation should be commented in the code using Doxygen format.

Reply:
This was roughly the best error obtained.
The `N=8` case hovers around 1e-5 to 3e-5 when `reltol=1e-7; abstol=1e-7`. This the point where the error on the `N=8` system "bottoms out". Higher and lower tolerances increase error. `error_tol` was chosen based of this.

Reply:
I think how you came up to the tolerance used for testing needs to be documented in the code. The argument for selecting that specific tolerance is unconvincing. Eventually we will want to improve this test and it would be good to have documented what has been done to this point.

Also, I would put numerical value for the tolerance and the comments in the `SolutionKeys.hpp` file.

Reply:
Added documentation on why it was chosen. Results are commented in showing why as well.